### ISSUE TITLE
Collection source null check

### ISSUE THREAD

**Issue by wenerme** [mapstruct/mapstruct#865]
```java

// mapper for member field
        if ( e.getMembers() != null ) {

            e.getMembers().clear();
            // may add null to target collection, check s.getMembers() != null is what I expected
            e.getMembers().addAll( projMemberDTOListToProjCoreUserEntityList( s.getMembers() ) );
        }

        else {

            e.setMembers( projMemberDTOListToProjCoreUserEntityList( s.getMembers() ) );
        }

// mapper source collection
    protected List<ProjCoreUserEntity> projMemberDTOListToProjCoreUserEntityList(List<ProjMemberDTO> list) {

        if ( list == null ) {

            return null;
        }

        List<ProjCoreUserEntity> list_ = new ArrayList<ProjCoreUserEntity>();

        for ( ProjMemberDTO projMemberDTO : list ) {

            list_.add( map( projMemberDTO ) );
        }

        return list_;
    }
```

**Comment by wenerme** [comment]
For every collection field I have to assign an empty collection.

**Comment by gunnarmorling** [comment]
Hi @wenerme, it's not quite clear to me what the issue is you are reporting. Could you describe more clearly what's the behavior you'd expect from MapStruct and what's the behavior you currently get instead? Thanks!

**Comment by agudian** [comment]
And maybe also check this section of the docs: http://mapstruct.org/documentation/1.1/reference/html/index.html#mapping-result-for-null-arguments

**Comment by agudian** [comment]
Now I understand:
* You have an update-method (mapping method with `@MappingTarget` parameter).
* A collection property needs to be mapped and for the element types there is a mapping method - the iterable mapping method is generated automatically (we call that a _forged method_ internally).
* The generated iterable mapping method returns `null` by default if the source is `null`. But as the result is directly passed to `addAll(..)`, you get a `NullPointerException`.

The right thing to do would be to perform a null-check on the result of the property-mapping method (user-specified or not) and then only call `addAll(..)` if it's not null.

**Comment by wenerme** [comment]
Sorry for weekend delay, @agudian yes, I don't expected `addAll(null)`, this will cause NPE (JPA Persist Collection).

**Comment by sjaakd** [comment]
Fixed with #878. @agudian : thanks for reviewing.
